### PR TITLE
More unit tests - cover more use cases

### PR DESCRIPTION
**Pull request by Sybrand**
- More unit tests to cover more use cases

Lots of gross fixture file and json. Sorry.

# Test Links:
[Percentile Calculator](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1761.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: api/app/tests/hfi/test_hfi_endpoint_request.py [hunk 1]
```diff
@@ -8,7 +8,38 @@
...
+                station_code=station_code, planning_area_id=planning_area_id)
+            fuel_type_code = get_fuel_type_code_by_station_code(station_code)
+            fuel_type = FuelType(abbrev=fuel_type_code, fuel_type_code=fuel_type_code,
+                                 description=fuel_type_code,
+                                 percentage_conifer=100, percentage_dead_fir=0)
+            result.append((planning_station, fuel_type))
+        return result
+
+    monkeypatch.setattr(app.utils.hfi_calculator, 'get_all_stations', mock_get_all_stations)
+    monkeypatch.setattr(app.routers.hfi_calc, 'get_fire_centre_stations', mock_get_fire_centre_stations)
```
Comment (JulianForeman) [inline api/app/tests/hfi/test_hfi_endpoint_request.py:42]: What is monkeypatch?

**Review by JulianForeman — APPROVED** [review]

**Comment by Sybrand** [comment]
> Have we verified if the calculated responses are correct?

I've eyeballed them. My intention is more around regression testing. If the output changes, I want to be very intentional about it.

This approach does get tricky though, since you have to take care when changing the API contract, not to let bugs in the calculation sneak in.

File: api/app/tests/hfi/test_hfi_endpoint_request.feature [hunk 2]
```diff
@@ -7,8 +7,11 @@ Feature: /hfi/
         And the response is <response_json>
 
         Examples:
-            | request_json                   | status_code | response_json                   |
+            # TODO: These test currently exposes a "bug" in the code where removing a station from one area, means it's removed from all.
```
Comment (andrea-williams) [inline api/app/tests/hfi/test_hfi_endpoint_request.feature:10]: Do you mean when a station is deselected in one fire centre it gets deselected in all other fire centres that use that same station?

**Review by andrea-williams — APPROVED** [review]

File: api/app/tests/hfi/test_hfi_endpoint_request.py (hunk 1 above)
Comment (conbrad) [reply api/app/tests/hfi/test_hfi_endpoint_request.py:42]: And mocks/stubs are described here: https://martinfowler.com/articles/practical-test-pyramid.html#MockingAndStubbing

File: api/app/tests/hfi/test_hfi_endpoint_request.feature (hunk 2 above)
Comment (Sybrand) [reply api/app/tests/hfi/test_hfi_endpoint_request.feature:10]: No, planning area, not fire centre.

**Pull request merged**